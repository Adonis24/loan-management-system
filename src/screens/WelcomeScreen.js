//console.ignoredYellowBox = ['Setting a timer']
import React, { useState, useEffect } from 'react';
import {
    Image,
    Text,
    TouchableOpacity,
    View,
    Animated,
    Easing,


} from 'react-native';
import { LinearGradient } from 'expo-linear-gradient'
import Constants from 'expo-constants'
//import { Constants, LinearGradient, FileSystem } from 'expo'

import Layout from '../constants/Layout'

//import { Ionicons } from '@expo/vector-icons';
import styles from '../styles/styles'

const WelcomeScreen = (props) => {

    const [logo, setLogo] = useState(new Animated.Value(0))
    const [intro, setIntro] = useState(new Animated.Value(0))
    const [buttons, setButtons] = useState(new Animated.Value(0))

    const animate = () => {

        Animated.stagger(1000, [
            Animated.timing(logo, {
                toValue: 1,
                duration: 2000,
                easing: Easing.linear
            }),
            Animated.timing(intro, {
                toValue: 1,
                duration: 2000,
                easing: Easing.linear
            }),
            Animated.timing(buttons, {
                toValue: 1,
                duration: 2000,
                easing: Easing.linear
            }),


        ]).start();

    }



    useEffect(() => {
        animate()
    }, []);



    const logoOpac = logo.interpolate({
        inputRange: [0, 1],
        outputRange: [0, 1,]
    })

    const introOpac = intro.interpolate({
        inputRange: [0, 1],
        outputRange: [0, 1,]
    })

    const buttonsOpac = buttons.interpolate({
        inputRange: [0, 1],
        outputRange: [0, 1,]
    })


    return (
        <View style={styles.container}>
            <View style={{ flex: 1, justifyContent: 'flex-start' }}>
                <Image source={require('../assets/images/tekunE.png')} style={{ width: Layout.window.width, height: Layout.window.height / 3.5 }} resizeMode={'cover'} />
            </View>
            <View style={{ position: 'absolute', top: 0, left: 0, bottom: 0, right: 0, }}>
                <View style={{ flex: 1, justifyContent: 'center', alignItems: 'center' }}>
                    <View style={{ width: Layout.window.width * 0.8, justifyContent: 'center', alignItems: 'center' }}>
                        <Animated.Image source={require('../assets/images/logo.png')} style={{ opacity: logoOpac, height: Layout.window.height * 0.2, width: Layout.window.width * 0.7 }} resizeMode={'contain'} />
                        <Animated.Text style={[styles.textDefault, { margin: 20, opacity: introOpac }]}>Welcome to Tent! Sign up now to join us or login to your account</Animated.Text>

                        <Animated.View style={{ opacity: buttonsOpac }}>
                            <TouchableOpacity onPress={() => props.navigation.navigate('Registration')} style={[styles.box,{ width: Layout.window.width * 0.4, borderWidth: 1, borderColor: '#4A90E2' }]}>
                                <Text style={[styles.textDefault, { color: '#4A90E2' }]}>Sign Up</Text>
                            </TouchableOpacity>
                            <TouchableOpacity onPress={() => props.navigation.navigate('Login')} >
                                <LinearGradient
                                    colors={['#4DCB3E', '#269B1D']}
                                    style={[styles.box,{width: Layout.window.width * 0.4,borderColor:'#4A90E2',borderWidth:1,borderRadius:15 }]}>
                                    <Text style={[styles.textDefault, { color: '#fff' }]}>Log In</Text>
                                </LinearGradient>
                            </TouchableOpacity>
                        </Animated.View>
                    </View>
                </View>
            </View>
        </View >
    );
}


export default WelcomeScreen