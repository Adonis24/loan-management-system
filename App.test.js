import React from 'react';
import renderer from 'react-test-renderer';
import WelcomeScreen from './src/screens/WelcomeScreen';

describe('<WelcomeScreen />', () => {
 
  it('renders correctly', () => {
    const tree = renderer.create(<WelcomeScreen />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});