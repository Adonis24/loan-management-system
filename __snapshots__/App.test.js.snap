// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> renders correctly 1`] = `null`;

exports[`<WelcomeScreen /> renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff",
      "flex": 1,
      "paddingTop": 44,
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
        "justifyContent": "flex-start",
      }
    }
  >
    <Image
      resizeMode="contain"
      source={1}
      style={
        Object {
          "width": 750,
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "bottom": 0,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "justifyContent": "center",
            "width": 600,
          }
        }
      >
        <Image
          resizeMode="contain"
          source={1}
          style={
            Object {
              "height": 266.8,
              "opacity": 0,
              "width": 525,
            }
          }
        />
        <Text
          style={
            Object {
              "alignSelf": "center",
              "color": "#000",
              "fontFamily": "Montserrat_medium",
              "fontSize": 14,
              "margin": 20,
              "opacity": 0,
              "textAlign": "center",
            }
          }
        >
          Welcome to Tent! Sign up now to join us or login to your account
        </Text>
        <View
          style={
            Object {
              "opacity": 0,
            }
          }
        >
          <View
            accessible={true}
            focusable={true}
            isTVSelectable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "borderColor": "#4A90E2",
                "borderRadius": 15,
                "borderWidth": 1,
                "justifyContent": "center",
                "margin": 10,
                "opacity": 1,
                "paddingBottom": 5,
                "paddingTop": 5,
                "width": 300,
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "alignSelf": "center",
                    "color": "#000",
                    "fontFamily": "Montserrat_medium",
                    "fontSize": 14,
                    "textAlign": "center",
                  },
                  Object {
                    "color": "#4A90E2",
                  },
                ]
              }
            >
              Sign Up
            </Text>
          </View>
          <View
            accessible={true}
            focusable={true}
            isTVSelectable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <ViewManagerAdapter_ExpoLinearGradient
              proxiedProperties={
                Object {
                  "colors": Array [
                    4283288382,
                    4280720157,
                  ],
                  "endPoint": undefined,
                  "locations": undefined,
                  "startPoint": undefined,
                }
              }
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "borderRadius": 15,
                    "justifyContent": "center",
                    "margin": 10,
                    "paddingBottom": 5,
                    "paddingTop": 5,
                    "width": 225,
                  },
                  Object {
                    "borderColor": "#4A90E2",
                    "borderRadius": 15,
                    "borderWidth": 1,
                    "width": 300,
                  },
                ]
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "alignSelf": "center",
                      "color": "#000",
                      "fontFamily": "Montserrat_medium",
                      "fontSize": 14,
                      "textAlign": "center",
                    },
                    Object {
                      "color": "#fff",
                    },
                  ]
                }
              >
                Log In
              </Text>
            </ViewManagerAdapter_ExpoLinearGradient>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
